ykrt/src/compile/jitc_yk/jit_ir/parser.rs

```diff
@@ -305,18 +311,38 @@ impl<'lexer, 'input: 'lexer> JITIRParser<'lexer, 'input, '_> {
                             .parse::<u32>()
                             .map_err(|e| self.error_at_span(tiidx, &e.to_string()))?;
                         assert_eq!(self.m.tilocs.len(), usize::try_from(off).unwrap());
-                        match type_ {
-                            ASTType::Int(_) | ASTType::Ptr => {
-                                self.m
-                                    .push_tiloc(VarLocation::Register(Register::GP(Rq::RBX)));
+                        let type_ = self.process_type(type_)?;
+                        let size = self.m.type_(type_).byte_size().ok_or_else(|| {
+                            self.error_at_span(
+                                tiidx,
+                                "Assigning a trace input to a zero-sized type is nonsensical",
+                            )
+                        })?;
+                        match self.m.type_(type_) {
+                            Ty::Void => unreachable!(),
+                            Ty::Integer(_) | Ty::Ptr | Ty::Func(_) => {
+                                self.m.push_tiloc(yksmp::Location::Register(
+                                    reg_off,
+                                    u16::try_from(size).unwrap(),
+                                    0,
+                                    0,
+                                ));
                             }
-                            ASTType::Float(_) | ASTType::Double(_) => {
-                                self.m
-                                    .push_tiloc(VarLocation::Register(Register::FP(Rx::XMM0)));
+                            Ty::Float(_) => {
+                                let size = u16::try_from(size).unwrap();
+                                ti_off = ti_off.next_multiple_of(u32::from(size));
+                                // FIXME: Why 6?!
```

Comment:
6 is the base pointer.

Reply:
I'd like this stuff to be fixed in another PR: I've (mostly) just copied this over.